src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/AzureTableTransactionalStateStorage.cs

```diff
@@ -232,106 +243,128 @@ private bool FindState(long sequenceId, out int pos)
 
         private async Task<KeyEntity> ReadKey()
         {
-            var query = new TableQuery<KeyEntity>()
-                .Where(AzureTableUtils.PointQuery(this.partition, KeyEntity.RK));
-            TableQuerySegment<KeyEntity> queryResult = await table.ExecuteQuerySegmentedAsync(query, null).ConfigureAwait(false);
-            return queryResult.Results.Count == 0
-                ? new KeyEntity() { PartitionKey = this.partition }
-                : queryResult.Results[0];
+            var queryResult = table.QueryAsync<KeyEntity>(AzureTableUtils.PointQuery(this.partition, KeyEntity.RK)).ConfigureAwait(false);
+            await foreach (var result in queryResult)
+            {
+                return result;
+            }
+
+            return new KeyEntity()
+            {
+                PartitionKey = partition,
+                RowKey = KeyEntity.RK
+            };
```

Comment:
PointQuery shouldn't be necessary:
```suggestion
            => await table.GetEntityAsync<KeyEntity>(this.partition, KeyEntity.RK).ConfigureAwait(false);
```

Reply:
Doesn't GetEntityAsync throw when the item isn't found? We want it to return a new item instead in that case

Reply:
Ah, that makes sense now that I look at the code again. Yes, GetEntity throws when not found.